Zend/zend_compile.c

```diff
@@ -4709,7 +4709,13 @@ static void zend_compile_static_call(znode *result, zend_ast *ast, uint32_t type
 		if (opline->op1_type == IS_CONST) {
 			zend_string *lcname = Z_STR_P(CT_CONSTANT(opline->op1) + 1);
 			ce = zend_hash_find_ptr(CG(class_table), lcname);
-			if (!ce && CG(active_class_entry)
+			if (ce) {
+				if (ZEND_USER_CODE(ce->type)
+				 && (CG(compiler_options) & ZEND_COMPILE_IGNORE_OTHER_FILES)
+				 && (ce->info.user.filename != CG(active_op_array)->filename)) {
```

Comment:
``ZEND_USER_CODE`` is useful for functions. Its better to check for ``ZEND_INTERNAL_CLASS``.
It also makes sense to add check for ``ZEND_COMPILE_IGNORE_INTERNAL_CLASSES`` at the same time.

Reply:
> ZEND_USER_CODE is useful for functions.

Oh, sorry,. Dumb mistake.

Reply:
Since this code was duplicated in multiple places, I extracted it into a function.